src/certs/ca/cert/mod.rs

```diff
@@ -102,6 +109,34 @@ impl codicon::Encoder<Body> for Certificate {
     }
 }
 
+impl<'de> Deserialize<'de> for Certificate {
+    fn deserialize<D>(deserializer: D) -> std::result::Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        use codicon::Decoder;
+
+        let bytes: Vec<u8> = Deserialize::deserialize(deserializer)?;
+        Self::decode(bytes.as_slice(), ()).map_err(serde::de::Error::custom)
+    }
+}
+
+impl Serialize for Certificate {
+    fn serialize<S>(&self, serializer: S) -> std::result::Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        use std::slice::from_raw_parts;
+
+        let bytes = unsafe { from_raw_parts(self as *const Self as *const u8, size_of::<Self>()) };
+        let mut seq = serializer.serialize_seq(Some(bytes.len()))?;
+        for b in bytes {
+            seq.serialize_element(b)?;
+        }
+        seq.end()
```

Comment:
```suggestion
        serializer.serialize_bytes(bytes)
```

Reply:
I am unsure as to why using `serialize_bytes` precludes successful deserialization. My main theory is that `serialize_bytes` encodes these as a definite length array as opposed to an indefinite length array in the CBOR spec.